src/runtime/parse-property-value.ts

```diff
@@ -24,32 +24,63 @@ import { isComplexType, MEMBER_FLAGS } from '@utils';
  * @param propType the type of the prop, expressed as a binary number
  * @returns the parsed/coerced value
  */
-export const parsePropertyValue = (propValue: any, propType: number): any => {
-  // ensure this value is of the correct prop type
+export const parsePropertyValue = (propValue: unknown, propType: number): any => {
+  /**
+   * Allow hydrate parameters that contain a simple object, e.g.
+   * ```ts
+   * import { renderToString } from 'component-library/hydrate';
+   * await renderToString(`<car-detail car=${JSON.stringify({ year: 1234 })}></car-detail>`);
+   * ```
+   * @deprecated
+   */
+  if (
+    isComplexType(propValue) &&
+    typeof propValue === 'string' &&
+    ((propValue.startsWith('{') && propValue.endsWith('}')) || (propValue.startsWith('[') && propValue.endsWith(']')))
+  ) {
+    try {
+      return JSON.parse(propValue);
+    } catch (e) {
+      /* ignore */
+    }
+  }
+
+  /**
+   * Allow hydrate parameters that contain a complex non-serialized values.
+   */
+  if (typeof propValue === 'string' && propValue.startsWith(SERIALIZED_PREFIX)) {
+    return deserializeProperty(propValue);
```

Comment:
I'm slightly concerned this will mean that everyone will get the contents of these utils in their runtime bundle? 
1) it's not an insignificant amount of bytes 2) it's only relevant for those using SSR wdyt?